x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/datafeed/DatafeedUpdate.java

```diff
@@ -0,0 +1,331 @@
+    private final String jobId;
+    private final TimeValue queryDelay;
+    private final TimeValue frequency;
+    private final List<String> indices;
+    private final List<String> types;
+    private final QueryBuilder query;
+    private final AggregatorFactories.Builder aggregations;
+    private final List<SearchSourceBuilder.ScriptField> scriptFields;
+    private final Integer scrollSize;
+    private final ChunkingConfig chunkingConfig;
+
+    private DatafeedUpdate(String id, String jobId, TimeValue queryDelay, TimeValue frequency, List<String> indices, List<String> types,
+                           QueryBuilder query, AggregatorFactories.Builder aggregations, List<SearchSourceBuilder.ScriptField> scriptFields,
+                           Integer scrollSize, ChunkingConfig chunkingConfig) {
+        this.id = id;
+        this.jobId = jobId;
+        this.queryDelay = queryDelay;
+        this.frequency = frequency;
+        this.indices = indices;
+        this.types = types;
+        this.query = query;
+        this.aggregations = aggregations;
+        this.scriptFields = scriptFields;
+        this.scrollSize = scrollSize;
+        this.chunkingConfig = chunkingConfig;
+    }
+
+    /**
+     * Get the id of the datafeed to update
+     */
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(DatafeedConfig.ID.getPreferredName(), id);
+        addOptionalField(builder, Datafeed.JOB_ID, jobId);
+        if (queryDelay != null) {
+            builder.field(DatafeedConfig.QUERY_DELAY.getPreferredName(), queryDelay.getStringRep());
+        }
+        if (frequency != null) {
+            builder.field(DatafeedConfig.FREQUENCY.getPreferredName(), frequency.getStringRep());
+        }
+        addOptionalField(builder, DatafeedConfig.INDICES, indices);
+        addOptionalField(builder, DatafeedConfig.TYPES, types);
+        addOptionalField(builder, DatafeedConfig.QUERY, query);
+        addOptionalField(builder, DatafeedConfig.AGGREGATIONS, aggregations);
+        if (scriptFields != null) {
+            builder.startObject(DatafeedConfig.SCRIPT_FIELDS.getPreferredName());
+            for (SearchSourceBuilder.ScriptField scriptField : scriptFields) {
+                scriptField.toXContent(builder, params);
+            }
+            builder.endObject();
+        }
+        addOptionalField(builder, DatafeedConfig.SCROLL_SIZE, scrollSize);
+        addOptionalField(builder, DatafeedConfig.CHUNKING_CONFIG, chunkingConfig);
+        builder.endObject();
+        return builder;
+    }
+
+    private void addOptionalField(XContentBuilder builder, ParseField field, Object value) throws IOException {
+        if (value != null) {
+            builder.field(field.getPreferredName(), value);
+        }
+    }
+
+    String getJobId() {
```

Comment:
All of these `getters` were package private in core, do we think they should be kept that way in this package?

Reply:
If they're package private then they're basically only useful for the unit tests.  I don't think the unit tests use them, so they're redundant as they stand.

I don't think it's likely that an end user would want to query them, but equally I don't think it hurts, and it's more hassle for us if we get a request to add them back in the future, so let's keep them but make them `public`.